File: AHYEON/14.programmers/pro_다음큰숫자.swift
import Foundation

func solution(_ n:Int) -> Int
{
    var answer:Int = n + 1

    let oneCount = String(n, radix: 2).filter { $0 == "1"}.count

    while true {
        if oneCount == String(answer, radix: 2).filter{$0 == "1"}.count {
            break
        }
        answer += 1
    }

    return answer
}

File: 염성훈/14.Programmers_lv2_3/pro_다음 큰 숫자.py
def solution(n):

    #이진법으로 바꾸고 1의 갯수를 샌다.
    c = bin(n).count('1')

    #n이후 값들에서 가장 먼저 1의 count값과 같은 값이 나오면 m을 리턴한다.
    for m in range(n + 1 , 1000001):
        if bin(m).count('1') == c:
            return m

n = 78
print(solution(n))